api/src/main/java/org/apache/iceberg/actions/GenerateSymlinkManifest.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.actions;
+
+public interface GenerateSymlinkManifest
+    extends Action<GenerateSymlinkManifest, GenerateSymlinkManifest.Result> {
+
+  GenerateSymlinkManifest ignoreDeleteFiles();
```

Comment:
can we actually not ignore delete files for symlink tables? If this is always true then we don't need this config.

Reply:
Good point, with the current state of the procedure we don't need to have a user specify ignore delete files, because it isn't even supported. We can just fail in the execution if we detect delete files.

Reply:
I think we don't need to fail, it's just expected behavior of symlink tables to not have MOR deletes.